### ISSUE TITLE
CSV import send_file + short living tempfiles + X-Sendfile

### ISSUE THREAD

**Issue by leikind** [leikind/wice_grid#51]
CSV import send_file + short living tempfiles + X-Sendfile= problem

send_file in Rails 3 by default uses headers to have the web server send the file. Thus, this is a separate HTTP request and by that time the tempfile in which the CSV data might already have been deleted.

 Commenting line config.action_dispatch.x_sendfile_header = "X-Sendfile" forces send_file to send the file the old way, from the same request, but this is not a good solution because it forces all other calls to send_file to do so.

**Comment by leikind** [comment]
fixed in version 3.0.0

**Comment by chrisbloom7** [comment]
Can you point me to the commit that you fixed this in? I'm have a similar problem in a different gem and I would love to point the issue that I'm inevitably going to open to a suggested work around.

**Comment by leikind** [comment]
Here is the commit: https://github.com/leikind/wice_grid/commit/58b5c9533a852b32681e0d977d30af96a6e13a37

I spent some time studying the Rails code in order to find a way to use different techniques for sending files per request, and arrived at the conclusion that with the current architecture this is not possible. So what I did is just taking send_file from Rails 2 and using it from inside of WiceGrid. I don't like this approach and I think that a better way would be to find a way to make temporary files live longer, long enough for the second request caused by X-Sendfile.
